tests/install_upgrade_operators/network_policy/test_network_policy_components.py

```diff
@@ -0,0 +1,289 @@
+    server_container_spec = {
+        **POD_CONTAINER_SPEC,
+        "name": "test-server",
+        "command": [
+            "/bin/bash",
+            "-c",
+            f'while true; do echo -e "HTTP/1.1 200 OK\\n\\nServer response $(date)" | '
+            f"nc -l -p {TEST_SERVER_PORT}; done",
+        ],
+        "ports": [{"containerPort": TEST_SERVER_PORT, "protocol": "TCP"}],
+    }
+
+    server_pod = ServerPod(
+        name="network-policy-server-pod",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        security_context=POD_SECURITY_CONTEXT_SPEC,
+        containers=[server_container_spec],
+    )
+
+    with server_pod as pod:
+        pod.wait_for_status(status=Pod.Status.RUNNING)
+        yield pod
+
+
+@pytest.fixture
+def deployed_server_service(admin_client, hco_namespace, deployed_server_pod):  # noqa: ARG001
+    """Create a service to expose the server pod on given port"""
+    with Service(
+        name="network-policy-server-service",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        selector={"app": TEST_SERVER_APP_LABEL},
+        ports=[{"name": "http", "port": TEST_SERVER_PORT, "targetPort": TEST_SERVER_PORT, "protocol": "TCP"}],
+    ) as service:
+        yield service
+
+
+@pytest.fixture
+def network_policy_match_labels(network_policy_by_name):
+    match_labels = network_policy_by_name.instance.spec.get("podSelector", {}).get("matchLabels")
+    assert match_labels, f"NetworkPolicy {network_policy_by_name.name} has no/empty podSelector.matchLabels"
+    return dict(match_labels)
+
+
+@pytest.fixture
+def service_by_network_policy(admin_client, hco_namespace, network_policy_by_name, network_policy_match_labels):
+    """Find Service whose spec.selector is a subset of the NetworkPolicy podSelector.matchLabels."""
+    matching_services = []
+    for service in Service.get(dyn_client=admin_client, namespace=hco_namespace.name):
+        selector = service.instance.spec.get("selector", {})
+        # Service selector must be contained in the pod labels set
+        if all(network_policy_match_labels.get(k) == v for k, v in selector.items()):
+            matching_services.append(service)
+
+    assert len(matching_services) == 1, (
+        f"Expected exactly 1 Service whose spec.selector is a subset of NetworkPolicy {network_policy_by_name.name} "
+        f"podSelector in namespace {hco_namespace.name}, but found {len(matching_services)}: "
+        f"{[s.name for s in matching_services]}"
+    )
+
+    return matching_services[0]
+
+
+@pytest.fixture
+def service_ip(service_by_network_policy):
+    """Extract clusterIP for testing network policy enforcement."""
+    service_ip = service_by_network_policy.instance.spec.get("clusterIP")
+    assert service_ip, f"Service {service_by_network_policy.name} has no clusterIP"
+    if ":" in service_ip:
+        pytest.fail(f"This test does not support sending traffic to IPv6: {service_ip}")
```

Comment:
1. check based on `ipFamilies`
2. in dual stack you should have an ipv4, so the test should not run on single stack ipv6, right?

Reply:
@rlobillo you resolved this comment so i did not see your input
please use `ipv6_supported_cluster`